timemachine/fe/mcgregor.py

```diff
@@ -128,6 +128,30 @@ def __init__(self, n_vertices, edges):
             for edge_idx in edges:
                 self.ve_matrix[vertex_idx][edge_idx] = True
 
+    def _verify_partially_connected(self, source, node_states, n_mapped_nodes):
+        """A fast BFS node generator, adapted from _plain_bfs in networkx."""
```

Comment:
Would be good to explain in a bit more detail of what is being done by this function.

If I understand correctly, given a source verify that all of the other mapped nodes can be found by BFS.

Reply:
DEMAPPED vs UNVISITED seems pretty subtle in their difference? It seems to be distinguished  where demapped means you can ignore the node where unvisited allows traversal?

Does this mean that if you have a unvisited, but not mapped node, that connects two sets of mapped atoms that it is considered partially connected?

Reply:
If a node has been visited: it can be either `MAPPED` or explicitly `DEMAPPED`. Otherwise, it is `UNVISITED`.

> Does this mean that if you have a unvisited, but not mapped node, that connects two sets of mapped atoms that it is considered partially connected?

Yes.

Reply:
So the intention is to say that the core that is being evaluated *could* produce a connected core and not that it is currently a connected core?

Reply:
correct.